### DISCUSSION TITLE
Restoring a deleted codespace

### DISCUSSION THREAD

**Discussion by LiaMaggiorotto** [community/community#20162] [Codespaces] [answered]
Yesterday while working in a codespace, I committed my work as usual but failed to push my changes up to my branch. Because our codespaces error out when restarted, I deleted the codespace I was working on and started a new one this morning. It wasn't until I opened my new codespace that I realized all of my work from the previous day was missing. Is there a way to recover those commits or recover that deleted codespace?

**Comment by LiaMaggiorotto** [comment, marked as answer]
For those of you looking for this same issue, please open a support ticket with git hub. They are able to manually restore your deleted code space but do give them some time to resolve on your behalf. Hopefully in the future they will be able to roll something out that will enable us to restore codespaces previously deleted ourselves.

**Reply by pmyjavec** [reply]
I hope they can restore mine, just lost work too, raise d support ticket.

**Reply by OkolieOKS234** [reply]
So sad that after I committed my changes in my repo, everything disappeared when I deleted the repo.

**Comment by SDPreetham** [comment]
Is there any way to restore a codespace deleted due to the end of retention period? If so, please describe
